hack/jenkins/e2e.sh

```diff
@@ -39,9 +39,62 @@ function join_regex_no_empty() {
     fi
 }
 
+# Print an error message and exit with status 1.
+function error_exit() {
+  echo "$1" 1>&2
+  exit 1
+}
+
+# Retry a command up to a certain numbers.
```

Comment:
We do a similar thing [here](https://github.com/kubernetes/kubernetes/blob/master/hack/jenkins/upload-to-gcs.sh#L44-L55), and all over the place in the go code. No need to share with them, though.